### PR TITLE
ISPN-5949 Added jar:file scheme support when loading infinispan.xml

### PR DESCRIPTION
**Pull request by blocha**
https://issues.jboss.org/browse/ISPN-5949

### CODE REVIEW

File: commons/src/main/java/org/infinispan/commons/util/AbstractFileLookup.java [hunk 1]
```diff
@@ -60,7 +60,16 @@ public InputStream lookupFileStrict(String filename, ClassLoader cl) throws File
 
    @Override
    public InputStream lookupFileStrict(URI uri, ClassLoader cl) throws FileNotFoundException {
-      return new FileInputStream(new File(uri));
+      //in case we have not only file schema, but jar:file schema too, we have to get rid off all schemas
+      int startIndex = uri.toString().lastIndexOf(':');
+      String fileName = uri.toString().substring(startIndex + 1);
+
+      InputStream in = cl.getResourceAsStream(fileName);
```
Comment (slaskawi) [inline commons/src/main/java/org/infinispan/commons/util/AbstractFileLookup.java]: You could just `return cl.getResourceAsStream(fileName);`

File: jcache/embedded/src/test/resources/infinispan_uri.xml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<infinispan xmlns="urn:infinispan:config:7.2">
+
+    <cache-container default-cache="default">
+        <local-cache name="foo" statistics="true">
+            <eviction max-entries="200"/>
```
Comment (slaskawi) [inline jcache/embedded/src/test/resources/infinispan_uri.xml]: You might remove `eviction` and `expiration` - they have nothing in common with initializing procedure.

File: jcache/embedded/src/test/resources/infinispan_uri.xml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<infinispan xmlns="urn:infinispan:config:7.2">
```
Comment (slaskawi) [inline jcache/embedded/src/test/resources/infinispan_uri.xml]: Could you also change the namespace version to `9.0` (which is being developed on `master branch`).

File: jcache/embedded/src/test/java/org/infinispan/jcache/JCacheManagerUriNotHierarchicalTest.java [hunk 4]
```diff
@@ -0,0 +1,65 @@
+package org.infinispan.jcache;
```
Comment (slaskawi) [inline jcache/embedded/src/test/java/org/infinispan/jcache/JCacheManagerUriNotHierarchicalTest.java]: Could you please move those 2 tests into `JCacheConfigurationTest` ? After all, specifying configuration file is a configuration thing :)

File: jcache/embedded/src/test/java/org/infinispan/jcache/JCacheManagerUriNotHierarchicalTest.java [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+   public void testJCacheManagerWherePathContainsJarFileSchema() throws IOException {
+      withCachingProvider(provider -> {
+         //given
+         CacheManager manager = new JCacheManager(
+               URI.create("jar:file:infinispan_uri.xml"),
+               provider.getClass().getClassLoader(),
+               provider,
+               null);
+         Cache fooCache = manager.getCache("foo");
+         String keyName = "wrong";
```
Comment (slaskawi) [inline jcache/embedded/src/test/java/org/infinispan/jcache/JCacheManagerUriNotHierarchicalTest.java]: Do we need to obtain a cache? Perhaps creating `JCacheManager` instance would be enough for this test?

**Comment by slaskawi** [comment]
Great job @blocha! There are some minor things to be fixed but the code in general looks very nice. When applying the changes - could you please [squash all 3 commits](https://ariejan.net/2011/07/05/git-squash-your-latests-commits-into-one/)?

**Comment by blocha** [comment]
Thanks, code is fixed and commits are squashed :)

**Comment by slaskawi** [comment]
Looks good to me; @galderz any comments?

**Comment by galderz** [comment]
@slaskawi Checking

**Comment by galderz** [comment]
Integrated, thx @blocha

**Pull request closed without merging**